kakao/src/main/kotlin/com/agoda/kakao/Views.kt

```diff
@@ -337,21 +337,69 @@ class KTextInputLayout : KBaseView<KTextInputLayout>, TextInputLayoutAssertions
  * @see ScrollViewActions
  * @see BaseAssertions
  *
- * @param builder ViewBuilder which will match your list view
- * @param itemTypeBuilder Lambda with receiver where you pass your item providers
- *
  * @see KAdapterItem
  * @see KAdapterItemTypeBuilder
  */
-class KListView(builder: ViewBuilder.() -> Unit, itemTypeBuilder: KAdapterItemTypeBuilder.() -> Unit)
-    : ScrollViewActions, BaseAssertions {
+class KListView : ScrollViewActions, BaseAssertions {
+    val matcher: Matcher<View>
+    val itemTypes: Map<KClass<out KAdapterItem<*>>, KAdapterItemType<KAdapterItem<*>>>
+
+    override val view: ViewInteraction
+
+    /**
+     * Constructs view class with view interaction from given ViewBuilder
+     *
+     * @param function ViewBuilder which will result in view's interaction
+     * @param itemTypeBuilder Lambda with receiver where you pass your item providers
+     *
+     * @see ViewBuilder
+     */
+    constructor(builder: ViewBuilder.() -> Unit, itemTypeBuilder: KAdapterItemTypeBuilder.() -> Unit) {
+        val vb = ViewBuilder().apply(builder)
+        matcher = vb.getViewMatcher()
+        view = vb.getViewInteraction()
+        itemTypes = KAdapterItemTypeBuilder().apply(itemTypeBuilder).itemTypes
+    }
 
-    private val builder = ViewBuilder().apply(builder)
+    /**
+     * Constructs view class with parent and view interaction from given ViewBuilder
+     *
+     * @param parent Matcher that will be used as parent in isDescendantOfA() matcher
+     * @param function ViewBuilder which will result in view's interaction
+     * @param itemTypeBuilder Lambda with receiver where you pass your item providers
+     *
+     * @see ViewBuilder
+     */
+    constructor(parent: Matcher<View>, builder: ViewBuilder.() -> Unit,
+                itemTypeBuilder: KAdapterItemTypeBuilder.() -> Unit) : this({
+        isDescendantOfA { withMatcher(parent) }
+        builder(this)
+    }, itemTypeBuilder)
 
-    val matcher = this.builder.getViewMatcher()
-    val itemTypes = KAdapterItemTypeBuilder().apply(itemTypeBuilder).itemTypes
+    /**
+     * Constructs view class with parent and view interaction from given ViewBuilder
+     *
+     * @param parent DataInteraction that will be used as parent to ViewBuilder
+     * @param function ViewBuilder which will result in view's interaction
+     * @param itemTypeBuilder Lambda with receiver where you pass your item providers
+     *
+     * @see ViewBuilder
+     */
+    @Suppress("UNCHECKED_CAST")
+    constructor(parent: DataInteraction, builder: ViewBuilder.() -> Unit,
+                itemTypeBuilder: KAdapterItemTypeBuilder.() -> Unit) {
+        val makeTargetMatcher = DataInteraction::class.java.getDeclaredMethod("makeTargetMatcher")
```

Comment:
why do we need to use `getDeclaredMethod` of `makeTargetMatcher`?

Reply:
@cdsap Because this is the only way to get the final matcher out of the `DataInteraction`, since it's private. Then we use this matcher as parent when matching items of list and recycler.